hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/TargetType.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.model;
+
+import java.util.Set;
+
+/**
+ * A {@link TargetType} is an abstract definition for
+ * {@link Target}
+ *
+ */
+public interface TargetType extends NamedEntity {
+    /**
+     * Maximum length of color in Management UI.
+     */
+    int COLOUR_MAX_SIZE = 16;
+
+    /**
+     * @return immutable set of optional {@link DistributionSetType}s
+     */
+    Set<DistributionSetType> getCompatibleDistributionSetTypes();
+
```

Comment:
A method to retrieve all targets with this TargetType should be provided similar to distribution set types.
E.g.
```java
Set<Target> getTargets();
```